future/atomicTransactionComposer.go

```diff
@@ -572,19 +584,23 @@ func (atc *AtomicTransactionComposer) Execute(client *algod.Client, ctx context.
 
 	txinfo, err := WaitForConfirmation(client, atc.txContexts[indexToWaitFor].txID(), waitRounds, ctx)
 	if err != nil {
-		return 0, nil, nil, err
+		return ExecuteResult{}, err
 	}
 	atc.status = COMMITTED
 
-	var returnValues []ABIResult
+	executeResponse := ExecuteResult{
+		ConfirmedRound: txinfo.ConfirmedRound,
+		TxIDs:          atc.getTxIDs(),
+	}
```

Comment:
nit: okay to initialize the `MethodResults` slice with size of `TxIDs`, so no need for append later in loop?
i.e., `make([]ABIMethodResults, len(atc.getTxIDs()))` (just an example)